### ISSUE TITLE
Results of unit tests on raspi, python 3.7 vs python 3.9

### ISSUE THREAD

**Issue by ankith26** [pygame/pygame#2152]
I have the latest raspberry pi OS (buster), on my raspberry pi 3b.
Python version 3.7 is pre-installed, I have installed pygame 2.0.0dev12 using pip

I have also installed python v3.9 using pyenv and pygame 2.0.0dev12

SDL version is 2.0.9 in both cases.

Below I have pasted the last part of the output text of running pygame standard unittests.

For Python 3.9

```
======================================================================
FAIL: test_SRCALPHA (pygame.tests.blit_test.BlitTest)
SRCALPHA tests.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/blit_test.py", line 19, in test_SRCALPHA
    self.assertEqual(s.get_at((0, 0)), d.get_at((0, 0)))
AssertionError: (0, 0, 255, 0) != (0, 0, 255, 255)

======================================================================
FAIL: test_get_active_iconify (pygame.tests.display_test.DisplayModuleTest)
Test the get_active function after an iconify
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/display_test.py", line 112, in test_get_active_iconify
    self.assertEqual(pygame.display.get_active(), False)
AssertionError: True != False

======================================================================
FAIL: test_iconify (pygame.tests.display_test.DisplayModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/display_test.py", line 286, in test_iconify
    self.assertEqual(minimized_event, True)
AssertionError: False != True

======================================================================
FAIL: test_blanket_alpha (pygame.tests.surface_test.SurfaceSelfBlitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/surface_test.py", line 3280, in test_blanket_alpha
    self._assert_same(surf, comp)
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/surface_test.py", line 3195, in _assert_same
    self.assertEqual(
AssertionError: (255, 0, 0, 255) != (0, 255, 0, 255) : (255, 0, 0, 255) != (0, 255, 0, 255), bpp: 8

======================================================================
FAIL: test_colorkey (pygame.tests.surface_test.SurfaceSelfBlitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/surface_test.py", line 3258, in test_colorkey
    self._assert_same(surf, comp)
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/surface_test.py", line 3195, in _assert_same
    self.assertEqual(
AssertionError: (255, 0, 0, 255) != (0, 255, 0, 255) : (255, 0, 0, 255) != (0, 255, 0, 255), bpp: 8

======================================================================
FAIL: test_pixel_alpha (pygame.tests.surface_test.SurfaceSelfBlitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/surface_test.py", line 3289, in test_pixel_alpha
    self._assert_same(surf, comp)
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/surface_test.py", line 3195, in _assert_same
    self.assertEqual(
AssertionError: (0, 255, 0, 255) != (0, 255, 0, 0) : (0, 255, 0, 255) != (0, 255, 0, 0), bpp: 32

======================================================================
FAIL: test_subsurface (pygame.tests.surface_test.SurfaceSelfBlitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/surface_test.py", line 3331, in test_subsurface
    self._assert_same(surf, comp)
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/surface_test.py", line 3195, in _assert_same
    self.assertEqual(
AssertionError: (0, 255, 0, 255) != (0, 255, 0, 0) : (0, 255, 0, 255) != (0, 255, 0, 0), bpp: 32

----------------------------------------------------------------------
Ran 1581 tests in 148.677s

FAILED (failures=7)
```

For Python 3.7

```
======================================================================
FAIL: test_SRCALPHA (pygame.tests.blit_test.BlitTest)
SRCALPHA tests.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.7/site-packages/pygame/tests/blit_test.py", line 19, in test_SRCALPHA
    self.assertEqual(s.get_at((0, 0)), d.get_at((0, 0)))
AssertionError: (0, 0, 255, 0) != (0, 0, 255, 255)

======================================================================
FAIL: test_blanket_alpha (pygame.tests.surface_test.SurfaceSelfBlitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.7/site-packages/pygame/tests/surface_test.py", line 3280, in test_blanket_alpha
    self._assert_same(surf, comp)
  File "/home/pi/.local/lib/python3.7/site-packages/pygame/tests/surface_test.py", line 3200, in _assert_same
    % (a.get_at((x, y)), b.get_at((x, y)), a.get_bitsize())
AssertionError: (255, 0, 0, 255) != (0, 255, 0, 255) : (255, 0, 0, 255) != (0, 255, 0, 255), bpp: 8

======================================================================
FAIL: test_colorkey (pygame.tests.surface_test.SurfaceSelfBlitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.7/site-packages/pygame/tests/surface_test.py", line 3258, in test_colorkey
    self._assert_same(surf, comp)
  File "/home/pi/.local/lib/python3.7/site-packages/pygame/tests/surface_test.py", line 3200, in _assert_same
    % (a.get_at((x, y)), b.get_at((x, y)), a.get_bitsize())
AssertionError: (255, 0, 0, 255) != (0, 255, 0, 255) : (255, 0, 0, 255) != (0, 255, 0, 255), bpp: 8

======================================================================
FAIL: test_pixel_alpha (pygame.tests.surface_test.SurfaceSelfBlitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.7/site-packages/pygame/tests/surface_test.py", line 3289, in test_pixel_alpha
    self._assert_same(surf, comp)
  File "/home/pi/.local/lib/python3.7/site-packages/pygame/tests/surface_test.py", line 3200, in _assert_same
    % (a.get_at((x, y)), b.get_at((x, y)), a.get_bitsize())
AssertionError: (0, 255, 0, 255) != (0, 255, 0, 0) : (0, 255, 0, 255) != (0, 255, 0, 0), bpp: 32

======================================================================
FAIL: test_subsurface (pygame.tests.surface_test.SurfaceSelfBlitTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.7/site-packages/pygame/tests/surface_test.py", line 3331, in test_subsurface
    self._assert_same(surf, comp)
  File "/home/pi/.local/lib/python3.7/site-packages/pygame/tests/surface_test.py", line 3200, in _assert_same
    % (a.get_at((x, y)), b.get_at((x, y)), a.get_bitsize())
AssertionError: (0, 255, 0, 255) != (0, 255, 0, 0) : (0, 255, 0, 255) != (0, 255, 0, 0), bpp: 32

----------------------------------------------------------------------
Ran 1605 tests in 146.138s

FAILED (failures=5)
```

**Comment by ankith26** [comment]
All the tests that failed on Raspberry pi python v3.7 also failed on python v3.9.

Additionally. these 2 tests failed only on python v3.9
```
======================================================================
FAIL: test_get_active_iconify (pygame.tests.display_test.DisplayModuleTest)
Test the get_active function after an iconify
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/display_test.py", line 112, in test_get_active_iconify
    self.assertEqual(pygame.display.get_active(), False)
AssertionError: True != False

======================================================================
FAIL: test_iconify (pygame.tests.display_test.DisplayModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pi/.pyenv/versions/3.9.0/lib/python3.9/site-packages/pygame/tests/display_test.py", line 286, in test_iconify
    self.assertEqual(minimized_event, True)
AssertionError: False != True
```

**Comment by MyreMylar** [comment]
The shared failed tests are likely related to this SDL bug:

https://bugzilla.libsdl.org/show_bug.cgi?id=4365

That should be fixed in SDL 2.0.14 when the arm SIMD extensions are disabled by default. Not sure what is up with `get_active()` but it has been a twitchy test historically.

**Comment by ankith26** [comment]
I have worked out a pr for get active

**Comment by ankith26** [comment]
Most of these failing tests are resolved (thanks to MyreMylar). Only 2 tests were failing, but those are now fixed with SDL 2.0.14. So closing this issue
